game-core/src/test/java/games/strategy/triplea/delegate/battle/MustFightBattleExecutablesTest.java

```diff
@@ -0,0 +1,1670 @@
+    when(attachment1.getTransportCapacity()).thenReturn(2);
+    when(attachment1.getIsSea()).thenReturn(true);
+
+    final Tuple<Unit, UnitAttachment> unitAndAttachment2 = newUnitAndAttachment();
+    final Unit unit2 = unitAndAttachment2.getFirst();
+    when(unit2.getOwner()).thenReturn(attacker);
+    final UnitAttachment attachment2 = unitAndAttachment2.getSecond();
+    when(attachment2.getMovement(defender)).thenReturn(0);
+    when(attachment2.getIsSea()).thenReturn(true);
+
+    when(battleSite.getUnits()).thenReturn(List.of(unit, unit2));
+
+    battle.setUnits(List.of(unit2), List.of(unit), List.of(), List.of(), defender, List.of());
+    final List<IExecutable> execs = battle.getBattleExecutables(true);
+
+    final int index = getIndex(execs, MustFightBattle.RemoveUndefendedTransports.class);
+    final IExecutable step = execs.get(index);
+
+    final IDelegateBridge delegateBridge = mock(IDelegateBridge.class);
+    step.execute(null, delegateBridge);
+
+    verify(delegateBridge, never()).addChange(any());
+    verify(battle, never()).remove(any(), any(), any(), eq(false));
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has a destroyer, WW2v2 is true, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is either, then attacker>defender>standard")
+  void firstStrikeOrderAttHasDestroyerDefHasDestroyerWW2v2TrueSneakAttackTrueFalse() {
+    assertThatFirstStrikeStepOrder(
+        givenFirstStrikeBattleSetup(true, true, true, true, true),
+        List.of(
+            FirstStrikeBattleStep.ATTACKER,
+            FirstStrikeBattleStep.DEFENDER,
+            FirstStrikeBattleStep.STANDARD));
+  }
+
+  private Tuple<MustFightBattle, List<IExecutable>> givenFirstStrikeBattleSetup(
+      final boolean attackerDestroyer,
+      final boolean defenderDestroyer,
+      final boolean ww2v2,
+      final boolean defendingSubsSneakAttack,
+      final boolean ignoreDefendingSubsSneakAttack) {
+    final MustFightBattle battle = spy(newBattle(WATER));
+    lenient().doNothing().when(battle).firstStrikeAttackersFire(any());
+    lenient().doNothing().when(battle).firstStrikeDefendersFire(any());
+
+    when(gameProperties.get(SUB_RETREAT_BEFORE_BATTLE, false)).thenReturn(false);
+    when(gameProperties.get(TRANSPORT_CASUALTIES_RESTRICTED, false)).thenReturn(false);
+    when(gameProperties.get(WW2V2, false)).thenReturn(ww2v2);
+    if (!ignoreDefendingSubsSneakAttack) {
+      when(gameProperties.get(DEFENDING_SUBS_SNEAK_ATTACK, false))
+          .thenReturn(defendingSubsSneakAttack);
+    }
+
+    final Unit attackerUnit = attackerDestroyer ? givenUnitDestroyer() : givenUnit();
+    final Unit defenderUnit = defenderDestroyer ? givenUnitDestroyer() : givenUnit();
+
+    battle.setUnits(
+        List.of(defenderUnit), List.of(attackerUnit), List.of(), List.of(), defender, List.of());
+    final List<IExecutable> execs = battle.getBattleExecutables(true);
```

Comment:
The IExecutable comes from a battle, do we really need the tuple? Could we just get the second value by calling `battle.getBattleExecutables(true)` directly?

Reply:
I don't remember why I had the tuple.  The tests were once a bit different so I probably had a reason and then after refactoring them, the reason disappeared but I still had the tuple legacy.  It can be simplified.